packages/ui/src/components/select/select.tsx

```diff
@@ -0,0 +1,570 @@
+        "data-[disabled]:bg-ui-bg-disabled data-[disabled]:text-ui-fg-disabled data-[placeholder]:data-[disabled]:text-ui-fg-disabled data-[disabled]:cursor-not-allowed",
+        { "h-8 px-2 py-2 text-sm": size === "small" },
+        { "pl-10": search },
+        className
+      )}
+      data-placeholder={!selectedItem}
+      disabled={disabled}
+      {...props}
+      {...toggleButtonProps}
+    >
+      <div>
+        {search && (
+          <div className="text-ui-fg-muted absolute left-3 flex h-5 w-5 items-center justify-center">
+            {<MagnifyingGlassMini />}
+          </div>
+        )}
+        {children}
+      </div>
+    </DropdownMenu.Trigger>
+  )
+})
+
+Trigger.displayName = "Select.Trigger"
+
+const TriggerIcon = ({
+  className,
+  children,
+  ...props
+}: React.HTMLAttributes<HTMLDivElement>) => (
+  <div
+    className={clx(
+      "text-ui-fg-muted group-data-[disabled]/trigger:text-ui-fg-disabled z-10 ml-auto",
+      className
+    )}
+    {...props}
+  >
+    {children ? children : <ChevronUpDown />}
+  </div>
+)
+
+TriggerIcon.displayName = "Select.TriggerIcon"
+
+type ValueProps = {
+  placeholder?: string
+  value?: string
+} & React.HTMLAttributes<HTMLDivElement>
+
+const Value = ({ children, placeholder = "", value }: ValueProps) => {
+  const {
+    selectedItem,
+    multi,
+    selectedItems,
+    clearSelectedItems,
+    search,
+    isOpen,
+    onSearch,
+  } = useSelectContext()
+
+  if (search && isOpen)
+    return <SearchInput onChange={(e) => onSearch(e.target.value)} />
+
+  // If children are a complex element, render them directly
+  if (children && typeof children !== "string") return <div>{children}</div>
+
+  // If there's a truthy value prop, render it
+  if (value) return <div>{value}</div>
+
+  // If neither a single item nor multiple items are selected, render placeholder
+  if (!selectedItem && !selectedItems?.length) return <div>{placeholder}</div>
+
+  // If we're in a multi select and have some selected items
+  if (multi && selectedItems)
+    return (
+      <div className="text-ui-fg-base">
+        <Badge
+          className="relative z-20 mr-1"
+          onClick={() => clearSelectedItems()}
+        >
+          {selectedItems.length} <XMarkMini />
+        </Badge>
+        Selected
+      </div>
+    )
+
+  // Else just render the label on the selected item
+  return <div>{selectedItem!.label}</div>
+}
+
+const Content = React.forwardRef<
+  React.ElementRef<typeof DropdownMenu.Content>,
+  React.ComponentPropsWithoutRef<typeof DropdownMenu.Content>
+>(({ children, className, ...props }, ref) => {
+  const innerRef = React.useRef<HTMLDivElement | null>(null)
+
+  React.useImperativeHandle<HTMLDivElement | null, HTMLDivElement | null>(
+    ref,
+    () => innerRef.current
+  )
+
+  const { getMenuProps, onScrollToBottom } = useSelectContext()
+
+  const { ref: menuRef, ...menuProps } = getMenuProps({ ref: innerRef })
```

Comment:
This should be combined with the innerRef, currently getting an error on each time you hover an item due to this